betty/wikipedia.py

```diff
@@ -110,53 +108,55 @@ class _Retriever:
     def __init__(
         self,
         http_client: aiohttp.ClientSession,
-        cache_directory_path: Path,
+        cache: FileCache,
         # Default to seven days.
         ttl: int = 86400 * 7,
     ):
-        self._cache_directory_path = cache_directory_path
-        self._cache_directory_path.mkdir(exist_ok=True, parents=True)
+        self._cache = cache
         self._ttl = ttl
         self._http_client = http_client
         self._rate_limiter = RateLimiter(200)
         self._images: dict[str, Image | None] = {}
 
-    async def _request(self, url: str, extension: str | None = None) -> Any:
-        cache_file_path = self._cache_directory_path / hashlib.md5(url.encode("utf-8")).hexdigest()
-        if extension:
-            cache_file_path = cache_file_path.with_suffix(f'.{extension}')
+    async def _fetch(self, url: str) -> bytes | None:

```

Comment:
Extract fetch functionality into a separate class, which may be private for now. Doing this should allow us to refine the exact cache requirements for it.